File: mangodefend-ml-server/app/src/modules/scans/enums.py
from enum import Enum


class ScanVerdict(str, Enum):
    """Vonis status keamanan file & verifikasi signature."""
    MALICIOUS = "MALICIOUS"
    PENDING_VERIFICATION = "PENDING_VERIFICATION"
    FALSE_POSITIVE = "FALSE_POSITIVE"
    BENIGN = "BENIGN"
    UNKNOWN = "UNKNOWN"


class ScanSource(str, Enum):
    """Sumber asal vonis hasil pemindaian."""
    REDIS_CACHE = "REDIS_CACHE"
    CLOUD_DB = "CLOUD_DB"
    ML_INFERENCE = "ML_INFERENCE"
    NOT_FOUND = "NOT_FOUND"
    OFFLINE_HEURISTIC = "OFFLINE_HEURISTIC"
    OFFLINE_SYNC = "OFFLINE_SYNC"


class DatasetSource(str, Enum):
    """Sumber pengumpulan dataset sampel ML."""
    AUTO_COLLECT = "AUTO_COLLECT"
    MANUAL_UPLOAD = "MANUAL_UPLOAD"

File: antivirus-mobile/app/lib/features/scan/domain/enums.dart
enum ScanVerdict {
  malicious('MALICIOUS'),
  benign('BENIGN'),
  unknown('UNKNOWN');

  final String value;
  const ScanVerdict(this.value);

  factory ScanVerdict.fromString(String? value) {
    switch (value?.toUpperCase()) {
      case 'MALICIOUS':
        return ScanVerdict.malicious;
      case 'BENIGN':
        return ScanVerdict.benign;
      default:
        return ScanVerdict.unknown;
    }
  }
}

enum ScanSource {
  redisCache('REDIS_CACHE'),
  cloudDb('CLOUD_DB'),
  mlInference('ML_INFERENCE'),
  localDb('LOCAL_DB'),
  heuristicEngine('HEURISTIC_ENGINE'),
  offlineHeuristic('OFFLINE_HEURISTIC'),
  offlineSync('OFFLINE_SYNC'),
  notFound('NOT_FOUND');

  final String value;
  const ScanSource(this.value);

  factory ScanSource.fromString(String? value) {
    switch (value?.toUpperCase()) {
      case 'REDIS_CACHE':
        return ScanSource.redisCache;
      case 'CLOUD_DB':
        return ScanSource.cloudDb;
      case 'ML_INFERENCE':
        return ScanSource.mlInference;
      case 'LOCAL_DB':
        return ScanSource.localDb;
      case 'HEURISTIC_ENGINE':
        return ScanSource.heuristicEngine;
      case 'OFFLINE_HEURISTIC':
        return ScanSource.offlineHeuristic;
      case 'OFFLINE_SYNC':
        return ScanSource.offlineSync;
      default:
        return ScanSource.notFound;
    }
  }
}

enum ScanType {
  file('upload'),
  folder('folder'),
  fullSystem('full_system'),
  apkDownload('apk_download'),
  fileDownload('file_download'),
  realtimeMonitoring('realtime_monitoring');

  final String value;
  const ScanType(this.value);
}
